### ISSUE TITLE
Not working. Console says: "tslint output: No valid rules have been specified" 

### ISSUE THREAD

**Issue by NL33** [SublimeLinter/SublimeLinter#997]
I have just downloaded Sublime Text 3, as well as SublimeLinter and tslint. Other downloaded packages: Typescript and Origami. I am working with Nativescript.

SublimeLinter is not working--it does not pick up any errors at all. Everything seemed to go fine in the download of both SublimeLinter and tslint. My package control shows SublimeLinter and SublimeLinter-contrib-tslint as being present. I have reviewed installation instructions and did not find anything further that seemed applicable to my case.

When I open a .ts file, I get this message:

`tslint output: No valid rules have been specified `

I closed all files, removed all packages, and reinstalled.  It continues to not work. Now, when I open a .ts file, I get this message in the console:

```
SublimeLinter: tslint output:
Warning: The 'no-unused-variable' rule requires type information. 
SublimeLinter: No match for <_sre.SRE_Pattern object at 0x1069051e0> 
```

System info:

Sublime Text 3, build 3143 (downloaded Feb 15)
SublimeLinter v 3.10.10
SublimeLinter-contrib-tslint v1.2.0
macOS High Sierra 10.13.3

**Comment by kaste** [comment]
Hi 👋 !

This does not seem to be an issue with SublimeLinter core, but with the plugin/linter you're using. The output you cite is the actual output from tslint. So, this part is at least working. You probably have to configure a tslint.json file, or have to tell via some settings where these file are located.

**Comment by NL33** [comment]
Thank you for the info. If the issue is indeed a lack of proper tslint.json file, are you aware of info on what is a valid json configuration that sublimelinter will recognize? 

For example, I have created a tslint.json file, set the configuration in sublimelinter settings to point to that file, and added a few rules to that file. However, I still get the same error: "No valid rules have been specified."

If the issue is a lack of proper rules, I am unaware how to locate a set of proper rules to insert.

I have reviewed the docs for tslint and sublimelinter-contrib-tslint, and I do not find specific guidance on the actual code to insert into the tslint.json file.

**Comment by kaste** [comment]
The first thing is always, does it work on the command line? And then, we should be able to get it to work within Sublime.

(If you get it to work on the command line, and the tslint plugin for SublimeLinter still does not work. You could try my PR over there https://github.com/lavrton/SublimeLinter-contrib-tslint/pull/26 with SublimeLinter 4 (beta))

**Comment by NL33** [comment]
Thanks. I have been able to get it to work now. General steps:
1) follow installation steps at https://palantir.github.io/tslint/.  '$ tslint --init' creates the necessary tslint.config file.
2) in 'SublimeLinter.sublime-settings --User', point to that config file, like the config_filename line below:
```
   "tslint": {
        "@disable": false,
        "args": [],
        "config_filename": "/Users/[$YOURNAME]/desktop/tslint.json",
        "excludes": []
    }
```

3) these steps enabled me to link SublimeLinter to tslint. The next step is to add in the relevant rules that you want in your tslint.config file. I have not found a standard recommendation on this for typescript, but (for typescript with angular2+) I have found success using the "rules" configuration from here:  https://github.com/NativeScript/nativescript-angular/blob/master/nativescript-angular/tslint.json

(I have not used codelyzer for the rules directory, so did not insert that line from the repo)
